src/Forms/Form.php

```diff
@@ -1483,7 +1483,7 @@ public function loadDataFrom($data, $mergeStrategy = 0, $fieldList = null)
      * @param FieldList $fieldList An optional list of fields to process.  This can be useful when you have a
      * form that has some fields that save to one object, and some that save to another.
      */
-    public function saveInto(DataObjectInterface $dataObject, $fieldList = null)
+    public function saveInto(ViewableData|DataObjectInterface $dataObject, $fieldList = null)
```

Comment:
We can't change the method signature of public API
```suggestion
    public function saveInto(DataObjectInterface $dataObject, $fieldList = null)
```

Reply:
Updated